tests/javascript_tests/test_view_controller.js

```diff
@@ -0,0 +1,330 @@
+  });
+
+  /**
+   *
+   * Test the resize, toJSON and fromJSON methods raise the appropriate errors.
+   *
+   */
+  test('Test resize, toJSON and fromJSON methods', function(){
+
+    var container = $('<div id="does-not-exist"></div>');
+    var controller = EmperorViewControllerABC(container, 'foo', 'bar');
+    throws(function(){
+      controller.resize(10, 10);
+    }, Error, 'Cannot call this abstract method');
+
+    throws(function(){
+      controller.fromJSON('{foo:11}');
+    }, Error, 'Cannot call this abstract method');
+
+    throws(function(){
+      controller.toJSON();
+    }, Error, 'Cannot call this abstract method');
+  });
+
+
+
+  module("EmperorAttributeABC", {
+
+    setup: function(){
+      sharedDecompositionViewDict = {}
+      var decomp;
+
+      // setup function
+      name = "pcoa";
+      ids = ['PC.636', 'PC.635'];
+      coords = [
+        [-0.276542, -0.144964, 0.066647, -0.067711, 0.176070, 0.072969,
+         -0.229889, -0.046599],
+        [-0.237661, 0.046053, -0.138136, 0.159061, -0.247485, -0.115211,
+         -0.112864, 0.064794]];
+      pct_var = [26.6887048633, 16.2563704022, 13.7754129161, 11.217215823,
+                 10.024774995, 8.22835130237, 7.55971173665, 6.24945796136];
+      md_headers = ['SampleID', 'LinkerPrimerSequence', 'Treatment', 'DOB'];
+      metadata = [['PC.636', 'YATGCTGCCTCCCGTAGGAGT', 'Control', '20070314'],
+                  ['PC.635', 'YATGCTGCCTCCCGTAGGAGT', 'Fast', '20071112']];
+      decomp = new DecompositionModel(name, ids, coords, pct_var, md_headers,
+                                      metadata);
+      var dv = new DecompositionView(decomp);
+      sharedDecompositionViewDict['pcoa'] = dv;
+
+      name = "biplot";
+      ids = ['tax_1', 'tax_2'];
+      coords = [
+        [-1, -0.144964, 0.066647, -0.067711, 0.176070, 0.072969,
+         -0.229889, -0.046599],
+        [-0.237661, 0.046053, -0.138136, 0.159061, -0.247485, -0.115211,
+         -0.112864, 0.064794]];
+      pct_var = [26.6887048633, 16.2563704022, 13.7754129161, 11.217215823,
+                 10.024774995, 8.22835130237, 7.55971173665, 6.24945796136];
+      md_headers = ['SampleID', 'Gram'];
+      metadata = [['tax_1', '1'],
+                  ['tax_2', '0']];
+      decomp = new DecompositionModel(name, ids, coords, pct_var, md_headers,
+                                      metadata);
+      dv = new DecompositionView(decomp);
+      sharedDecompositionViewDict['biplot'] = dv;
+
+      // Slickgrid
+      var columns = [
+        {id: "pc1", name: "pc1", field: "pc1"},
+        {id: "pc2", name: "pc2", field: "pc2"},
+        {id: "pc3", name: "pc3", field: "pc3"},
+      ];
+
+      var options = {
+        enableCellNavigation: true,
+        enableColumnReorder: false
+      };
+      var data = [];
+      data.push({'pc1':1, 'pc2':1, 'pc3':1});
+      data.push({'pc1':1, 'pc2':1, 'pc3':2});
+
+      grid = new Slick.Grid("#id1", data, columns, options);
```

Comment:
Hmmmm, I don't think it's good practice to create divs as part of
index.html, we should create them and destroy them as part of setup and
teardown.

On (Dec-01-15|19:52), mortonjt wrote:
> > +      // Slickgrid
> > +      var columns = [
> > +        {id: "pc1", name: "pc1", field: "pc1"},
> > +        {id: "pc2", name: "pc2", field: "pc2"},
> > +        {id: "pc3", name: "pc3", field: "pc3"},
> > +      ];
> > +
> > +      var options = {
> > +        enableCellNavigation: true,
> > +        enableColumnReorder: false
> > +      };
> > +      var data = [];
> > +      data.push({'pc1':1, 'pc2':1, 'pc3':1});
> > +      data.push({'pc1':1, 'pc2':1, 'pc3':2});
> > +
> > +      grid = new Slick.Grid("#id1", data, columns, options);
> 
> That's coming from one of the divs in `index.html`
> 
> On Tue, Dec 1, 2015 at 4:35 PM, Yoshiki Vázquez Baeza <
> notifications@github.com> wrote:
> 
> > In tests/javascript_tests/test_view_controller.js
> > <https://github.com/biocore/emperor/pull/428#discussion_r46362504>:
> >
> > > +      // Slickgrid
> > > +      var columns = [
> > > +        {id: "pc1", name: "pc1", field: "pc1"},
> > > +        {id: "pc2", name: "pc2", field: "pc2"},
> > > +        {id: "pc3", name: "pc3", field: "pc3"},
> > > +      ];
> > > +
> > > +      var options = {
> > > +        enableCellNavigation: true,
> > > +        enableColumnReorder: false
> > > +      };
> > > +      var data = [];
> > > +      data.push({'pc1':1, 'pc2':1, 'pc3':1});
> > > +      data.push({'pc1':1, 'pc2':1, 'pc3':2});
> > > +
> > > +      grid = new Slick.Grid("#id1", data, columns, options);
> >
> > Where is the element #id1 coming from?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/biocore/emperor/pull/428/files#r46362504>.
> >
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/emperor/pull/428/files#r46374789